File: backend/app/schemas/audit.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime
from typing import Optional, List

class AuditLogCreate(BaseModel):
    user: str
    action: str
    resource: str
    status: Optional[str] = "Success"
    details: Optional[str] = None

class AuditLogResponse(AuditLogCreate):
    model_config = ConfigDict(from_attributes=True)
    id: int
    timestamp: datetime

class AuditLogListResponse(BaseModel):
    total: int
    logs: List[AuditLogResponse]

File: frontend/src/types/audit.ts
export interface AuditLog {
  id: number;
  timestamp: string;
  user: string;
  action: string;
  resource: string;
  status: string;
  details?: string;
}

export interface AuditLogListResponse {
  total: number;
  logs: AuditLog[];
}
